### DISCUSSION TITLE
global format How do avoid blank lines without any information?

### DISCUSSION THREAD

**Discussion by ghost** [starship/starship#5650] [General]
### format(global)
```
format = """
$username$os$directory
$status$cmd_duration
$character"""
```
### python(Simple default configuration)
```
[python]
symbol ="🐍 "
```

### format(global) 
```
format = """
$username$os$directory
$status$cmd_duration
$character"""
```
**shell style**
<img width="372" alt="no 1" src="https://github.com/starship/starship/assets/118535944/7ff9e947-6810-4d07-a7c4-ea90eb25e5a7">

### format(global) 
```
format = """
$username$os$directory
$python
$status$cmd_duration
$character"""
```
**shell style (In a directory without python)**
<img width="300" alt="2" src="https://github.com/starship/starship/assets/118535944/5b597157-122e-4e80-879f-77bfbd840f2b">

**shell style (In a directory with python)**
<img width="300" alt="3" src="https://github.com/starship/starship/assets/118535944/7474db11-acc8-44d3-8ac7-9d2939373e90">

In the directory without python, I found an extra blank line.
Everything is fine in the directory with python.
How can I avoid the appearance of blank lines?
I tried replacing `$python` with `$all` in format.  but it will still show a blank line.

**Comment by mickimnet** [comment]
Hello 1trapbox,

you need to escape the carriage return in the format string with three quotes—see: [Default Prompt Format](https://starship.rs/config/#default-prompt-format).

Take care
Mick

**Reply by ghost** [reply]
```
format = """
$username$os$directory
$python
$status$cmd_duration
$character"""
```
i do. pls advice :) thanks

**Reply by jankatins** [reply]
I would try to add a backslash at the end of each (or just the ones you so not want) line to "escape" the line break.

If you want python output to appear on its own line, i would experiment with adding a line break in the python format string or the symbol (so it's within the brackets of the format string, so it only appears when it has content).

**Comment by Gremious** [comment]
I've been struggling with this as well. The problem is that what we're writing is basically:

```toml
format = """\n
$username$os$directory\n
$python\n
$status$cmd_duration\n
$character""
```

So if python is empty, the format ends up:

```toml
format = """\n
$username$os$directory\n
\n
$status$cmd_duration\n
$character""
```

Cause you write the python section on a new line.

~~As far as I know - starship doesn't do conditional formatting. Would love to be shown wrong.~~

Nvm we got conditional formatting see next comment!

---

~~Escaping the python newline would just merge it with the other line which is not desirable.~~

~~My best suggestion is doing something like `$time$python` so that there's always something on that line (the time). Then at least it doesn't look as bad.~~

**Comment by Gremious** [comment]
I've been a liar!'

Starship has [conditional format strings](https://starship.rs/config/#conditional-format-strings)

So all you have to do is:

```toml
format = """
$username$os$directory
($python\n)\
$status$cmd_duration
$character""
```

and if you're using `$all`, it won't be empty for conditionals cause it uses `$line_break`. So ether use it somewhere yourself, or do:

```toml
[line_break]
disabled = true
```
